discovery/azure/azure_test.go

```diff
@@ -18,11 +18,11 @@ import (
 	"testing"
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
-	"go.uber.org/goleak"
+	"github.com/prometheus/prometheus/util/testutil"
 )
 
 func TestMain(m *testing.M) {
-	goleak.VerifyTestMain(m)
+	testutil.TolerantVerifyLeak(m)
```

Comment:
why is that change needed now?

Reply:
I think it's only temporary... I'm not entirely sure because I didn't trace exactly which transitive dependencies are pulling in `opencensus` and `klog`, but something definitely is.

We're flipping the dependencies around, so instead of `config` and `discovery`/`discoverer` depending on `azure`, now `azure` depends on `config` and `discovery`/`discoverer`. Since the migration is incomplete, that means that `azure` is now pulling in all other non-migrated discovery implementations (including `kubernetes`, which is my guess on where we're picking up the background goroutines). Once the migration is complete `config` and `discovery` won't depend on any of the discovery implementations, so they'll be safe for each discovery implementation to depend on.

I can add a TODO to revert this when the other dependencies are removed.

Reply:
Oh yes I see. This now depends in k8s indirectly. Thanks!